fabric-tool-attribute-v1/src/main/java/net/fabricmc/fabric/api/tool/attribute/v1/ToolAttributeHolder.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.tool.attribute.v1;
+
+import com.google.common.collect.ImmutableSetMultimap;
+import com.google.common.collect.Multimap;
+
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.attribute.EntityAttributeModifier;
+import net.minecraft.item.ItemStack;
+
+/**
+ * Interface for adding various tool attributes to items.
+ */
+public interface ToolAttributeHolder {
+	Multimap<String, EntityAttributeModifier> EMPTY = ImmutableSetMultimap.of();
+
+	/**
+	 * @param stack The stack to check on.
+	 * @param user The current user of the tool, or none if there isn't any
+	 * @return The mining level of the item. 3 is equal to a diamond pick.
+	 */
+	//TODO: nullable on user once we have an official @Nullable annotation in
+	int getMiningLevel(ItemStack stack, LivingEntity user);
+
+	/**
+	 * @param stack The stack to check on.
+	 * @param user The current user of the tool, or none if there isn't any
+	 * @return The mining speed of the item. 8.0 is equal to a diamond pick.
+	 */
+	//TODO: nullable on user once we have an official @Nullable annotation in
+	float getMiningSpeed(ItemStack stack, LivingEntity user);
```

Comment:
Similarly to [Yarn#1069](https://github.com/FabricMC/yarn/issues/1069), `getMiningSpeed` does not return the speed of mining/breaking the block - it returns one factor of many that determine the mining speed.

Reply:
Until Yarn#1069 is resolved, this will stay `getMiningSpeed`.

Reply:
IMO Fabric API should try to use the best names it can and yarn can be updated to match.

Reply:
That's not been the standard up to this point, so that should probably be discussed.

Reply:
Reason being API can't change easily at all while Yarn can